### PR TITLE
Feature/add mypy

### PR DESCRIPTION
**Pull request by armarik**
Added mypy to this library and fixed any errors that mypy pointed out. I had to update some dependencies to use a recent version of mypy.

Also did a little "fix" regarding the PaginatedResponse model, since python 3.9 does not allow generic Named Tuples.

### CODE REVIEW

File: pyproject.toml
Comment (Luned) [inline pyproject.toml:1]: I am not quite sure why I would have this error, but you don't:
image

Either way, pretty simple to solve, just install the most recent version of types-requests and everything works after that :)

**Review by Luned — APPROVED** [review]
Just some small details for you to check out. Don't really have anything to comment regarding the typing themselves, everything looked good to me and if Mypy is not complaining, I assume that it is good xD
I am approving, but please just update the things I mentioned!

File: tests/prism/docker-compose.yml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 version: "3.3"
 services:
-  vdx-helper:
+  vdx-helper-test:
```
[the remark this answers is not in the document]
Comment (armarik) [reply tests/prism/docker-compose.yml:3]: Agreed! I'll change the naming to something more prism like

File: pyproject.toml
Comment (armarik) [reply pyproject.toml:1]: ah i know what happened; i installed it via mypy T_T with the suggested command there. I'll add it directly to the project toml

Comment (armarik) [reply pyproject.toml:1]: also I think I know what happened here; i was dumb and installed types-requests via mypy command. I'll add the lib directly in the project toml and it should work

**Pull request merged**